File: app/src/main/java/io/connorwyatt/flashcards/data/contracts/CategoryContract.java
package io.connorwyatt.flashcards.data.contracts;

import io.connorwyatt.flashcards.data.entities.BaseColumnsTimeline;

public class CategoryContract {
    public static final String TABLE_NAME = "category";

    public static final String TABLE_CREATE = "create table "
            + TABLE_NAME + "( "
            + Columns._ID + " integer primary key autoincrement, "
            + Columns._CREATED_ON + " integer not null, "
            + Columns._LAST_MODIFIED_ON + " integer not null, "
            + Columns.NAME + " text not null"
            + ");";

    private CategoryContract() {
    }

    public static class Columns implements BaseColumnsTimeline {
        public static final String NAME = "name";
    }

}

File: app/src/main/java/io/connorwyatt/flashcards/data/contracts/CategoryContract.kt
package io.connorwyatt.flashcards.data.contracts

import io.connorwyatt.flashcards.data.entities.legacy.BaseColumnsTimelineLegacy

@Deprecated("This is considered legacy.")
object CategoryContract
{
    val TABLE_NAME = "category"
    val TABLE_CREATE = "create table $TABLE_NAME(" +
                       "${BaseColumnsTimelineLegacy._ID} integer primary key autoincrement," +
                       "${BaseColumnsTimelineLegacy._CREATED_ON} integer not null," +
                       "${BaseColumnsTimelineLegacy._LAST_MODIFIED_ON} integer not null," +
                       "${Columns.NAME} text not null" +
                       ");"

    object Columns
    {
        val NAME = "name"
    }
}
